LibSerialize.lua

```diff
@@ -1265,10 +1279,13 @@ end
 
 function LibSerialize:SerializeEx(opts, ...)
     self:_ClearReferences()
```

Comment:
Since the library stores state on itself during serialization, you're going to have a bad time if someone (perhaps another addon) tries to serialize something while another serialization is happening asynchronously. You'll need to update all such state to be managed at the function call level. Also add a test case for this (make sure it fails in your current version).

Reply:
You can probably rename `combinedOpts` and use it as the state for the current serialization, since it's already plumbed everywhere. Just need to make sure the loop that writes user-specified keys is restricted to valid options.

Reply:
Do not store any global states in the library.

The state can be a function local variable, because the state is cleared at the beginning and at the end of Serialization/Deserialization,

Change the prototype of  self:_ClearReferences() to  self:_ClearReferences(state), where "state" is initialized as a table.
Store "refsDirty", "stringRefs" and "tableRefs" in the variable "state" instead.

Reply:
Yep, I probably shouldn't have done it that way to begin with, but it didn't make a difference until now :) unless people tried to dig into the library's internals, I suppose, but that was never a support scenario and I'm not concerned about breaking them.